scripts/globals/mobskills/soporific.lua

```diff
@@ -18,20 +18,19 @@ mobskillObject.onMobWeaponSkill = function(target, mob, skill)
         local resist = xi.mobskills.applyPlayerResistance(mob, nil, target, mob:getStat(xi.mod.INT)-target:getStat(xi.mod.INT), 1, xi.magic.ele.DARK)
         local duration = math.ceil(60 + math.floor(31 * math.random()) * resist) -- wiki: duration variable from 30 to 90. can be thought of random 60-90 with possible half resist making it range 30-90
         if resist >= 0.5 then
-            target:delStatusEffectSilent(effect)
-            target:delStatusEffectSilent(xi.effect.LULLABY)
-            target:delStatusEffectSilent(xi.effect.SLEEP_I)
-            target:delStatusEffectSilent(xi.effect.POISON)
-            local dotdmg = 50
-            if not (target:hasImmunity(xi.immunity.SLEEP) or hasSleepEffects(target)) and target:addStatusEffect(effect, 1, 0, duration, 25, 25, 1) then -- subid/subpower for poison detection on wakup function
-                target:addStatusEffect(xi.effect.POISON, dotdmg, 3, duration, 3, 15, 2)
+            if
+                not (target:hasImmunity(xi.immunity.SLEEP) or hasSleepEffects(target)) and
+                -- use nightmare tick with 50 hp every 3 seconds
+                target:addStatusEffect(effect, 20, 3, duration, 135, 50)
+            then
```

Comment:
Recent LSB PR I pulled down removed `hasSleepEffects` in order to pre-load enums so this needs to be changed back, sorry!

Reply:
> Recent LSB PR I pulled down removed `hasSleepEffects` in order to pre-load enums so this needs to be changed back, sorry!

Ok, I changed that file to use logic like other files where hasSleepEffects was removed, thanks!